src/core/test_utils.rs

```diff
@@ -0,0 +1,47 @@
+#[cfg(test)]
+macro_rules! test_owned_trait_requirements {
+    ($test_name:ident, $owned:ty, $ffi_type:ty) => {
+        #[test]
+        fn $test_name() {
+            use crate::ffi::sealed::{AsPtr, FromMutPtr};
+
+            fn assert_clone<T: Clone>() {}
+            #[allow(drop_bounds)]
```

Comment:
I think we can just skip the check for `drop`. If it is not implemented correctly, we leak memory, which hopefully either the sanitizers, or valgrind will pick up.